tests/secrets/env.test.ts

```diff
@@ -694,7 +721,40 @@ describe('commandEnv', () => {
     });
     expect(result.exitCode).toBe(0);
     const jsonOut = JSON.parse(result.stdout);
-    expect(jsonOut['SECRET']).toBe('this is a secret\nit has multiple lines\n');
+    // Remove last character
+    expect(jsonOut[secretName]).toBe(secretContent.slice(0, -1));
+  });
+  // TODO: fastcheck
+  test('trailing newlines are preserved with option', async () => {
+    const vaultId = await polykeyAgent.vaultManager.createVault(vaultName);
+    const secretName = 'SECRET';
+    const secretContent = 'this is a secret\n';
+    await polykeyAgent.vaultManager.withVaults([vaultId], async (vault) => {
+      await vaultOps.addSecret(vault, secretName, secretContent);
+    });
+    const command = [
+      'secrets',
+      'env',
+      '-np',
+      dataDir,
+      '--env-format',
+      'unix',
+      '--preserve-newline',
+      `${vaultName}:${secretName}`,
+      `${vaultName}:${secretName}`,
+      '--',
+      'node',
+      '-e',
+      'console.log(JSON.stringify(process.env))',
+    ];
```

Comment:
Very interestingly, if the order of the flag is changed to the following, the flag gets parsed as a vault name instead. I tried to replicate this behaviour on the command line manually, but that didn't work, and it worked as expected.

```ts
const command = [
  ...
  `${vaultName}:${secretName}`,
  '--preserve-newline',
  `${vaultName}:${secretName}`,
  ...
];
```

First, for the tests, specifying a single `'--'` works but shell needs two of them, now this inconsistency. Something weird is going on. I will investigate this when I work next.

Reply:
So, apparently, this issue is resolved by removing the `this.passThroughOptions()`, which makes the tests work as on the CLI, but that has the side-effect of the tests also needing the double `'--'` to be parsed properly. I think we might need to rely on `process.argv` to process the secret paths and the commands separately.

Reply:
Were the `--preserve-newline` options being set before the arguments of the command?  The comment above seems to suggest that you were setting the options after the arguments. Which `passThroughOptions` will treat as arguments because it's telling commander to treat everything after the first argument as more aguments.

Reply:
Interestingly enough, doing it like that actually works on the CLI, and the CLI needs two `'--'`, as if `passThroughOptions()` isn't having an effect on the actual CLI of the command, but just the tests.

If `passThroughOptions()` doesn't even work on the CLI, and only on the tests (as what we can see here), should I just remove it and look for alternative solutions for the `'--'` issue, like trying to use `process.argv` for arguments and parsing instead of `commander`?

Reply:
After digging into the reason for needing double `--` and the inconsistency between the tests and the CLI, I discovered that I needed an option, `this.enablePositionalOptions()` to be set. Without it, I cannot pass through the options and they are parsed as normal.

However, enabling the option breaks other things, especially the option availability on all subcommand layers. That doesn't work, and my tries of using ChatGPT to fix it have succeeded a little, but ultimately failed to fix all the issues.

I will make an issue on `commanderjs` repo, asking them to add a feature to allow this, or provide a workaround.

Reply:
Until then, to run commands, just use double `--` instead of a single one.

```console
[aryanj@matrix-34xx:~]$ polykey secrets env vault:SECRET -- -- bash -c 'echo $SECRET'
this is the secret contents
```